packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1237,39 +1237,42 @@ bool hasPlugins(FlutterProject project) {
 ///
 ///  For more details, https://flutter.dev/go/federated-plugins.
 // TODO(stuartmorgan): Expand implementation to apply to all implementations,
-// not just Dart-only, per the federated plugin spec.
+//  not just Dart-only, per the federated plugin spec.
+// TODO(gustl22): The flag [throwOnPluginPubspecError] is currently only used
+//  for testing dart plugins as the logic is not working correctly.
 List<PluginInterfaceResolution> resolvePlatformImplementation(
   List<Plugin> plugins, {
   bool throwOnPluginPubspecError = true,
 }) {
-  final List<String> platforms = <String>[
+  const Iterable<String> platformKeys = <String>[
     AndroidPlugin.kConfigKey,
     IOSPlugin.kConfigKey,
     LinuxPlugin.kConfigKey,
     MacOSPlugin.kConfigKey,
     WindowsPlugin.kConfigKey,
   ];
-  final Map<String, List<PluginInterfaceResolution>> possibleResolutions
-      = <String, List<PluginInterfaceResolution>>{};
+  final Map<String, List<PluginInterfaceResolution>> possibleResolutions =
+      <String, List<PluginInterfaceResolution>>{};
   final Map<String, String> defaultImplementations = <String, String>{};
   // Generates a key for the maps above.
   String getResolutionKey({required String platform, required String packageName}) {
     return '$packageName:$platform';
   }
 
   bool hasPubspecError = false;
-  for (final Plugin plugin in plugins) {
-    for (final String platform in platforms) {
-      if (plugin.platforms[platform] == null &&
-          plugin.defaultPackagePlatforms[platform] == null) {
+  for (final String platformKey in platformKeys) {
```

Comment:
why did you invert the order of these for loops?

Reply:
In the final implementation attempt, the same logic is used for resolving dart plugins as well as native plugins in [_finalPluginResolution](https://github.com/flutter/flutter/pull/137040/files#diff-68361f6fc9aafa5fa7de042eec4603dd4f321eb783a00edd57780ef78c1094bbR1329).
To achieve that the resolving is made first per each platform and then per plugin, which also makes more sense as the priority which plugin implementation is picked as the final one is also decided for each platform separately.